.github/workflows/build.yaml

```diff
@@ -9,6 +9,76 @@ on:
     branches: [ master ]
 
 jobs:
+  changes:
+    name: changed files
+    runs-on: ubuntu-latest
+    # Set job outputs to values from changed_files step for other jobs
+    outputs:
+      codecheck: ${{ steps.changed_files.outputs.codecheck }}
+      clang_tidy: ${{ steps.changed_files.outputs.clang_tidy }}
+      run_test_suite: ${{ steps.changed_files.outputs.test_suite }}
+      inst_windows_msvc: ${{ steps.changed_files.outputs.inst_windows_msvc }}
+      inst_windows: ${{ steps.changed_files.outputs.inst_windows }}
+      inst_macos: ${{ steps.changed_files.outputs.inst_macos }}
+      appimage: ${{ steps.changed_files.outputs.appimage }}
+    steps:
+    - name: Check for changed files
+      uses: dorny/paths-filter@v2
+      id: changed_files
+      with:
+        # groups to check for changed files (as string, yaml anchors are resolved by the tool)
+        filters: |
+          codecheck:
+            - 'CMakeLists.txt'
+            - '*/CMakeLists.txt'
+            - 'cmake/codecheck/**'
+            - 'cmake/WlFunctions.cmake'
+            - 'src/**'
+            - '.github/scripts/install_deps.sh' # calls install-dependencies.sh
+            - '.github/workflows/build_codecheck.yaml'
+          clang_tidy:
+            - &src_cmake_dependencies
+              - 'CMakeLists.txt'
+              - 'cmake/**'
+              - 'src/**'
+                # dependencies:
+              - 'install-dependencies.sh' # called by xx.yaml or ./.github/scripts/install_deps.sh
+              - 'utils/download_asio.sh' # called by install-dependencies.sh
+            - 'utils/check_clang_tidy_results.py'
+            - 'utils/run-clang-tidy.py'
+            - '.github/scripts/install_deps.sh' # calls install-dependencies.sh
+            - '.github/workflows/build_clang_tidy.yaml'
+          test_suite:
+            - *src_cmake_dependencies
+            - 'data/**' # TODO(aDiscoverer or ...) list what needs to be checked
+            - 'regression_test.py'
+            - 'test/**'
+            - '.github/scripts/install_deps.sh' # calls install-dependencies.sh
+            - '.github/workflows/build_testsuite.yaml'
+          inst_windows_msvc:
+            - &installer
+              - *src_cmake_dependencies
+              - 'data/**'
```

Comment:
add `utils/asio_not_packaged.sh`, or probably to some installers only. No idea which ones use it.
https://github.com/widelands/widelands/blob/897ab9fdd78a888f339ea0a64b356bb8d14e236c/install-dependencies.sh#L103

Reply:
It looks to me that only OpenBSD and the Solus and Void distros need `download_asio.sh`, neither of which is used in the workflow. So it can be removed from the checks.

Reply:
It already was in `src_cmake_dependencies`.
https://github.com/widelands/widelands/blob/c6522ff016211307224c366ced1b8916399bfb77/.github/workflows/build.yaml#L38
So it could be removed.

Reply:
I removed it, as I had to resolve a merge conflict anyway.